src/sentry/monitors/models.py

```diff
@@ -481,21 +485,84 @@ def mark_failed(self, last_checkin=None, reason=MonitorFailure.UNKNOWN):
         if not affected:
             return False
 
-        event_manager = EventManager(
-            {
-                "logentry": {"message": f"Monitor failure: {self.monitor.name} ({reason})"},
-                "contexts": {"monitor": get_monitor_environment_context(self)},
-                "fingerprint": ["monitor", str(self.monitor.guid), reason],
-                "environment": self.environment.name,
-                # TODO: Both of these values should be get transformed from context to tags
-                # We should understand why that is not happening and remove these when it correctly is
-                "tags": {"monitor.id": str(self.monitor.guid), "monitor.slug": self.monitor.slug},
-            },
-            project=Project(id=self.monitor.project_id),
-        )
-        event_manager.normalize()
-        data = event_manager.get_data()
-        insert_data_to_database_legacy(data)
+        group_type, level = get_group_type_and_level(reason)
+        current_timestamp = datetime.utcnow().replace(tzinfo=timezone.utc)
+
+        use_issue_platform = False
+        try:
+            organization = Organization.objects.get(id=self.monitor.organization_id)
+            use_issue_platform = features.has(
+                "organizations:issue-platform", organization=organization
+            ) and features.has("organizations:crons-issue-platform", organization=organization)
+        except Organization.DoesNotExist:
+            pass
+
+        if use_issue_platform:
+            from sentry.issues.issue_occurrence import IssueEvidence, IssueOccurrence
+            from sentry.issues.producer import produce_occurrence_to_kafka
+
+            occurrence = IssueOccurrence(
+                id=uuid.uuid4().hex,
+                resource_id=None,
+                project_id=self.monitor.project_id,
+                event_id=uuid.uuid4().hex,
+                fingerprint=[f"monitor-{str(self.monitor.guid)}-{reason}"],
+                type=group_type,
+                issue_title=f"Monitor failure: {self.monitor.name} ({reason})",
+                subtitle="",
+                evidence_display=[
+                    IssueEvidence(name="Failure reason", value=reason, important=True),
+                    IssueEvidence(name="Environment", value=self.environment.name, important=False),
+                    IssueEvidence(
+                        name="Last check-in", value=last_checkin.isoformat(), important=False
+                    ),
+                ],
+                evidence_data={},
+                culprit="",
+                detection_time=current_timestamp,
+                level=level,
+            )
+
+            produce_occurrence_to_kafka(
+                occurrence,
+                {
+                    "environment": self.environment.name,
+                    "event_id": occurrence.event_id,
+                    "platform": "crons",
```

Comment:
I'm not sure if `crons` will work as a platform, it's usually specific to the framework that the event was fired from.

Reply:
I'm not sure if empty works, but you could try it. Based on the docs you might want `other` https://develop.sentry.dev/sdk/event-payloads/